axum/src/serve.rs

```diff
@@ -235,18 +235,24 @@ where
 
                 let tcp_stream = TokioIo::new(tcp_stream);
 
-                poll_fn(|cx| make_service.poll_ready(cx))
-                    .await
-                    .unwrap_or_else(|err| match err {});
```

Comment:
This is actually fine because the error variant is `Infallible`, which is a type that cannot be constructed. It's defined as an enum with 0 variants, so there is no way to construct it. That's also partly why the `match` is here, if anyone adds a variant to that enum for some reason, this would not compile anymore.

Basically we know that the closure never actually runs. The compiler also knows it and can optimize all this code away. When you added the body here, it actually also is unreachable code.